providers/implementations/ciphers/ciphercommon.c

```diff
@@ -435,19 +441,29 @@ int ossl_cipher_generic_block_final(void *vctx, unsigned char *out,
         return 0;
     }
 
-    if (ctx->pad && !ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz)) {
-        /* ERR_raise already called */
-        return 0;
+    pad = ~constant_time_is_zero(ctx->pad);
+
+    if (pad) {
+        gp = ~constant_time_is_zero(ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz));
     }
 
+    ret = constant_time_select(pad, gp, CONSTTIME_TRUE);
```

Comment:
I think this can be simplified to just:
```
int ret = 1;
[...]
if (ctx->pad)
    ret = ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz);
[...]
return ret;
```

I don't see a reason why all the constant time functions should get called. The ossl_cipher_unpadblock() already does this, and we don't do anything with ossl_cipher_unpadblock()'s return value other than return it.

Also note that your style guide requires the removal of the `{` and `}`.